### PR TITLE
Makes for loops use GLOB.human_list. Should improve performance a bit. Fixes blood contract affecting dead crew

### PR DESCRIPTION
**Pull request by farie82**
## What Does This PR Do
Makes all for loops that use `GLOB.mob_list` or such use `GLOB.human_list` instead.

`var/mob/living/carbon/human/.* in GLOB` was used to search the loops and such

- The `GLOB.alive_mob_list` replacements have an alive check.
- The `GLOB.player_list` replacements have a client check

The blood contract now won't make dead mobs want to kill somebody.... this didn't make much sense. I'll comment the line involved.

## Why It's Good For The Game
Should improve performance a bit.
The blood contract thing was just a weird thing I came across

## Changelog
:cl:
fix: The blood contract now won't make dead people want to hunt you.
/:cl:

### CODE REVIEW

File: code/modules/mining/lavaland/loot/bubblegum_loot.dm [hunk 1]
```diff
@@ -81,8 +81,9 @@
 			var/obj/effect/mine/pickup/bloodbath/B = new(L)
 			B.mineEffect(L)
 
-		for(var/mob/living/carbon/human/H in GLOB.player_list)
-			if(H == L)
+		for(var/thing in GLOB.human_list)
+			var/mob/living/carbon/human/H = thing
+			if(!H.client || H.stat == DEAD || H == L)
```
Comment (farie82) [inline code/modules/mining/lavaland/loot/bubblegum_loot.dm:86]: Here is the gameplay change

File: code/controllers/subsystem/afk.dm [hunk 2]
```diff
@@ -21,7 +21,8 @@ SUBSYSTEM_DEF(afk)
 
 /datum/controller/subsystem/afk/fire()
 	var/list/toRemove = list()
-	for(var/mob/living/carbon/human/H in GLOB.alive_mob_list)
+	for(var/thing in GLOB.human_list)
+		var/mob/living/carbon/human/H = thing
```
Comment (Fox-McCloud) [inline code/controllers/subsystem/afk.dm:25]: This is going to trigger on dead mobs now; functionality change; not sure if we should do this.

File: code/controllers/subsystem/ticker.dm [hunk 3]
```diff
@@ -397,8 +397,9 @@ SUBSYSTEM_DEF(ticker)
 
 /datum/controller/subsystem/ticker/proc/equip_characters()
 	var/captainless=1
-	for(var/mob/living/carbon/human/player in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/controllers/subsystem/ticker.dm:400]: player_list is pretty distinctively different from the human list. I'm not sure how good of an idea this is.

File: code/datums/datacore.dm [hunk 4]
```diff
@@ -222,8 +222,10 @@ GLOBAL_LIST_EMPTY(PDA_Manifest)
 
 
 /datum/datacore/proc/manifest()
-	for(var/mob/living/carbon/human/H in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/datums/datacore.dm:225]: Again.

File: code/controllers/subsystem/afk.dm (hunk 2 above)
Comment (farie82) [reply code/controllers/subsystem/afk.dm:25]: It should remove them from the watch list if they are in there due to line 33. `H.stat || ...`. Or they will just be ignored if they are not in the watch list

File: code/game/gamemodes/game_mode.dm [hunk 5]
```diff
@@ -282,7 +282,8 @@
 
 /datum/game_mode/proc/num_players_started()
 	. = 0
-	for(var/mob/living/carbon/human/H in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/game_mode.dm:285]: same.

File: code/game/gamemodes/cult/cult_objectives.dm [hunk 6]
```diff
@@ -154,13 +154,15 @@
...
 	var/list/possible_sac_targets = list()
-	for(var/mob/living/carbon/human/player in GLOB.player_list)
-		if(player.mind && !is_convertable_to_cult(player.mind) && (player.stat != DEAD) && (!player.mind.offstation_role) )
+	for(var/thing in GLOB.human_list)
+		var/mob/living/carbon/human/player = thing
+		if(player.client && player.mind && !is_convertable_to_cult(player.mind) && (player.stat != DEAD) && (!player.mind.offstation_role) )
 			possible_sac_targets += player.mind
 	if(!possible_sac_targets.len)
 	//There are no living Unconvertables on the station. Looking for a Sacrifice Target among the ordinary crewmembers
-		for(var/mob/living/carbon/human/player in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/cult_objectives.dm:162]: same.

File: code/game/gamemodes/cult/cult_objectives.dm [hunk 7]
```diff
@@ -154,13 +154,15 @@
 
 /datum/game_mode/cult/proc/get_possible_sac_targets()
 	var/list/possible_sac_targets = list()
-	for(var/mob/living/carbon/human/player in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/cult_objectives.dm:157]: same.

File: code/game/gamemodes/cult/cult.dm [hunk 8]
```diff
@@ -230,8 +230,9 @@ GLOBAL_LIST_EMPTY(all_cults)
 
 /datum/game_mode/cult/proc/get_unconvertables()
 	var/list/ucs = list()
-	for(var/mob/living/carbon/human/player in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/cult.dm:233]: same.

File: code/game/gamemodes/blob/blob.dm [hunk 9]
```diff
@@ -58,8 +58,9 @@ GLOBAL_LIST_EMPTY(blob_nodes)
 
 /datum/game_mode/blob/proc/get_blob_candidates()
 	var/list/candidates = list()
-	for(var/mob/living/carbon/human/player in GLOB.player_list)
-		if(!player.stat && player.mind && !player.client.skip_antag && !player.mind.special_role && !jobban_isbanned(player, "Syndicate") && (ROLE_BLOB in player.client.prefs.be_special))
+	for(var/thing in GLOB.human_list)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/blob/blob.dm:61]: same.

File: code/game/gamemodes/intercept_report.dm [hunk 10]
```diff
@@ -109,7 +109,8 @@
 
 /datum/intercept_text/proc/get_suspect()
 	var/list/dudes = list()
-	for(var/mob/living/carbon/human/man in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/intercept_report.dm:112]: same.

File: code/game/gamemodes/objective.dm [hunk 11]
```diff
@@ -471,7 +471,8 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 						continue
 					n_p++
 		else if(SSticker.current_state == GAME_STATE_PLAYING)
-			for(var/mob/living/carbon/human/P in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/objective.dm:474]: same.

File: code/game/gamemodes/wizard/rightandwrong.dm [hunk 12]
```diff
@@ -161,7 +161,10 @@ GLOBAL_VAR_INIT(summon_magic_triggered, FALSE)
 	else
 		CRASH("Bad summon_type given: [summon_type]")
 
-	for(var/mob/living/carbon/human/H in GLOB.player_list)
+	for(var/thing in GLOB.human_list)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/wizard/rightandwrong.dm:164]: same.

File: code/modules/admin/topic.dm [hunk 13]
```diff
@@ -2708,7 +2711,8 @@
 				feedback_inc("admin_secrets_fun_used",1)
 				feedback_add_details("admin_secrets_fun_used","TA([objective])")
 
-				for(var/mob/living/carbon/human/H in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/modules/admin/topic.dm:2711]: same.

File: code/modules/admin/topic.dm [hunk 14]
```diff
@@ -2816,7 +2820,8 @@
 			if("stupify")
 				feedback_inc("admin_secrets_fun_used",1)
 				feedback_add_details("admin_secrets_fun_used","RET")
-				for(var/mob/living/carbon/human/H in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/modules/admin/topic.dm:2819]: same.

File: code/modules/admin/verbs/one_click_antag.dm [hunk 15]
```diff
@@ -57,8 +57,9 @@ client/proc/one_click_antag()
 	log_admin("[key_name(owner)] tried making [antnum] traitors with One-Click-Antag")
 	message_admins("[key_name_admin(owner)] tried making [antnum] traitors with One-Click-Antag")
 
-	for(var/mob/living/carbon/human/applicant in GLOB.player_list)
```
Comment (Fox-McCloud) [inline code/modules/admin/verbs/one_click_antag.dm:60]: same.

File: code/modules/admin/verbs/one_click_antag.dm [hunk 16]
```diff
@@ -88,8 +89,9 @@ client/proc/one_click_antag()
 	log_admin("[key_name(owner)] tried making [antnum] changelings with One-Click-Antag")
 	message_admins("[key_name_admin(owner)] tried making [antnum] changelings with One-Click-Antag")
 
-	for(var/mob/living/carbon/human/applicant in GLOB.player_list)
-		if(CandCheck(ROLE_CHANGELING, applicant, temp))
+	for(var/thing in GLOB.human_list)
```
Comment (Fox-McCloud) [inline code/modules/admin/verbs/one_click_antag.dm:92]: same.

File: code/controllers/subsystem/ticker.dm (hunk 3 above)
Comment (farie82) [reply code/controllers/subsystem/ticker.dm:400]: I'll change these back!

File: code/modules/admin/verbs/one_click_antag.dm [hunk 17]
```diff
@@ -118,8 +120,9 @@ client/proc/one_click_antag()
 	log_admin("[key_name(owner)] tried making [antnum] revolutionaries with One-Click-Antag")
 	message_admins("[key_name_admin(owner)] tried making [antnum] revolutionaries with One-Click-Antag")
 
-	for(var/mob/living/carbon/human/applicant in GLOB.player_list)
-		if(CandCheck(ROLE_REV, applicant, temp))
+	for(var/thing in GLOB.human_list)
```
Comment (Fox-McCloud) [inline code/modules/admin/verbs/one_click_antag.dm:123]: same. You get the idea. Leave player_list alone.

**Pull request merged**